extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -47,9 +49,17 @@ class AmpCarousel extends AMP.BaseElement {
       'goToSlide',
       actionInvocation => {
         const {args, trust} = actionInvocation;
-        this.carousel_.goToSlide(args['index'] || 0, {
-          actionSource: this.getActionSource_(trust),
-        });
+        const slide = Number(args['index'] || 0);
+        if (isNaN(slide)) {
```

Comment:
Since the error is contingent upon a conditional, let's replace `user().error()` with `userAssert()`:
```js
userAssert(
  !isNaN(slide), 
  'Unexpected slide index for goToSlide action: %s. %s',
  args['index'], 
  this.element);
```

Reply:
Done. Thanks!